src/Akka.Streams.Kafka.Tests/BugFix240SupervisionStrategy.cs

```diff
@@ -0,0 +1,434 @@
+                .PlainSource(settings, Subscriptions.Assignment(new TopicPartition(topic1, 0)))
+                .WithAttributes(ActorAttributes.CreateSupervisionStrategy(Decider))
+                .Select(c => c.Value)
+                .RunWith(this.SinkProbe<int>(), Materializer);
+
+            probe.Request(elementsCount);
+            probe.ExpectNoMsg(TimeSpan.FromSeconds(10));
+            callCount.Should().Be(elementsCount);
+            probe.Cancel();
+        }        
+        
+        [Fact]
+        public async Task SupervisionStrategy_Decider_on_PlainSource_should_stop_on_internal_error()
+        {
+            int elementsCount = 10;
+            var topic1 = CreateTopic(1);
+            var group1 = CreateGroup(1);
+
+            var sourceTask = ProduceStrings(new TopicPartition(topic1, 0), Enumerable.Range(1, elementsCount), ProducerSettings);
+
+            await GuardWithTimeoutAsync(sourceTask, TimeSpan.FromSeconds(3));
+            
+            var settings = CreateConsumerSettings<Null, string>(group1).WithAutoCreateTopicsEnabled(false);
+
+            var probe = KafkaConsumer
+                .PlainSource(settings, Subscriptions.Assignment(new TopicPartition(topic1, 5)))
+                .Select(c => c.Value)
+                .RunWith(this.SinkProbe<string>(), Materializer);
+
+            probe.Request(elementsCount);
+            var error = probe.ExpectEvent(TimeSpan.FromSeconds(5));
+            error.Should().BeOfType<TestSubscriber.OnError>();
+            var exception = ((TestSubscriber.OnError)error).Cause;
+            exception.Should().BeOfType<KafkaException>();
+            ((KafkaException) exception).Error.Code.Should().Be(ErrorCode.Local_UnknownPartition);
+            
+            probe.Cancel();
+        }
+        
+        [Fact]
+        public async Task PlainSource_should_stop_on_errors()
+        {
+            int elementsCount = 10;
+            var topic1 = CreateTopic(1);
+            var group1 = CreateGroup(1);
+
+            await ProduceStrings(new TopicPartition(topic1, 0), Enumerable.Range(1, elementsCount), ProducerSettings);
+
+            var settings = CreateConsumerSettings<int>(group1).WithValueDeserializer(Deserializers.Int32);
+
+            var probe = KafkaConsumer
+                .PlainSource(settings, Subscriptions.Assignment(new TopicPartition(topic1, 0)))
+                .WithAttributes(ActorAttributes.CreateSupervisionStrategy(Deciders.StoppingDecider))
+                .Select(c => c.Value)
+                .RunWith(this.SinkProbe<int>(), Materializer);
+
+            var error = probe.Request(elementsCount).ExpectEvent(TimeSpan.FromSeconds(5));
+            error.Should().BeOfType<TestSubscriber.OnError>();
+            var exception = ((TestSubscriber.OnError)error).Cause;
+            exception.Should().BeOfType<ConsumeException>();
+            ((ConsumeException) exception).Error.IsSerializationError().Should().BeTrue();
```

Comment:
They do terminate the stream, but they left the KafkaConsumerActor running, which is kind of weird, because they did not implement any supervision decider on their consumer/source stage. Their default is to crash the stage if it happens and let the stages downstream to pick up the error and handle it.

Reply:
Can one `KafkaConsumerActor` get used by multiple stages in the event of a `SubSource`?

Reply:
Yes, KafkaConsumerActor is shared with all of the SubSource, they watch the actor and fails their stage if the actor died

Reply:
And the actor can only terminate itself is the `Consumer` throws a fatal exception? Or is it when the consumer throws  _any_ exception?

Reply:
In other words, can that 10/10 brokers down error cause this actor to die?

Reply:
Yes, it will die. But the decider in the stage will restart it.